contrib/lib/src/space_helmet/helmet.rs

```diff
@@ -0,0 +1,287 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use rocket::fairing::{Fairing, Info, Kind};
+use rocket::http::Header;
+use rocket::{Request, Response, Rocket};
+
+use super::policy::*;
+
+/// SpaceHelmet implements a [Fairing](../../rocket/fairing/trait.Fairing.html)
+/// that adds http headers to outgoing responses that control security features
+/// on the client.
+///
+/// # Usage
+///
+/// `SpaceHelmet` can be used in several ways.
+///
+/// To use it with its [defaults](#method.default), create a new instance and
+/// attach it to [Rocket](../../rocket/struct.Rocket.html)
+/// as shown.
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///
+/// use rocket_contrib::space_helmet::SpaceHelmet;
+///
+/// let helmet = SpaceHelmet::default();
+/// let rocket = rocket::ignite().attach(helmet);
+///
+/// ```
+///
+/// To enable an additional header, call the method for that header, with the
+/// policy for that header, before attach e.g.
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::{SpaceHelmet, ReferrerPolicy};
+///  let helmet = SpaceHelmet::default().referrer_policy(ReferrerPolicy::NoReferrer);
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// To disable a header, call the method for that header with `None` as
+/// the policy.
+/// ```rust
+///
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::SpaceHelmet;
+///  let helmet = SpaceHelmet::default().no_sniff(None);
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// `SpaceHelmet` supports the builder pattern to configure multiple policies
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::{HSTSPolicy, ExpectCTPolicy, ReferrerPolicy, SpaceHelmet};
+///  let helmet = SpaceHelmet::default()
+///     .hsts(HSTSPolicy::default())
+///     .expect_ct(ExpectCTPolicy::default())
+///     .referrer_policy(ReferrerPolicy::default());
+///
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// # TLS and HSTS
+///
+/// If tls is enabled when rocket
+/// [launched()'ed](../../rocket/fairing/trait.Fairing.html) in a
```

Comment:
Should this point to `Fairing`?

This is formatting oddly due to the links. I'd suggest splitting the reference from the anchor:

```
If TLS is enabled when a [`Rocket`] is launched...

[`Rocket`]: /rocket/struct.Rocket.html
```